axelor-account/src/main/java/com/axelor/apps/account/web/MoveTemplateController.java

```diff
@@ -114,14 +114,13 @@ public void generateMove(ActionRequest request, ActionResponse response) {
     }
   }
 
-  public void setIsValid(ActionRequest request, ActionResponse response) {
+  public void setIsValid(ActionRequest request, ActionResponse response){
     MoveTemplate moveTemplate = request.getContext().asType(MoveTemplate.class);
-    if (moveTemplate.getIsValid()) {
-      return;
-    }
-    for (MoveTemplateLine line : moveTemplate.getMoveTemplateLineList()) {
-      line.setIsValid(moveTemplate.getIsValid());
+    if(!moveTemplate.getIsValid()) {
+      for(MoveTemplateLine line : moveTemplate.getMoveTemplateLineList()) {
+        line.setIsValid(false);
+      }
+      response.setValue("moveTemplateLineList", moveTemplate.getMoveTemplateLineList());
     }
```

Comment:
Why increase complexity here ?

Reply:
I inverted the condition to remove the return which was not useful. Also the second moveTemplate.getIsValid() is always false because of the condition so no need to call it twice.